bin/cron/hoc_student_name_cleanup

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env ruby
+require_relative 'only_one'
+abort 'Script already running' unless only_one_running?(__FILE__)
+
+# This script deletes hoc_activity.name from all rows with a finished_at time
+# between the given start and end dates. It is designed to run every night as
+# a cron job.
+#
+# This script is part 2 of a two-step process to ensure that student names
+# stored in pegasus.hoc_activity are deleted after 3 months. This script assumes
+# that the finished_at value is a resonable estimate for when a name is written
+# to the row. This relationship is almost always true in practice for recently
+# written rows but it is not directly enforced by our code. (Older rows may not
+# meet this assumption and were removed by bin/oneoff/hoc_student_name_initial_cleanup.)
+#
+# The logic in this script was constrained by the current schema of the table
+# because the table is very large and changing its schema is difficult. When
+# online schema changes become available, we should add an indexed column to
+# track when the name is added to the table and update the logic in this script
+# accordingly.
+
+require File.expand_path('../../../pegasus/src/env', __FILE__)
+require 'optparse'
+require 'optparse/date'
+require 'cdo/db'
+require 'cdo/properties'
+require 'cdo/chat_client'
+
+script_start_time = Time.now
+CDO.log.info("Script started at #{script_start_time}")
```

Comment:
are these logs namespaced in some way? `CDO.log.info` sounds pretty global, so if this is logging to a general log for the whole app i'd consider prefixing your logging with the script name or similar.

Reply:
Good question.  In my testing, this just logs to console, but Suresh says there's some magic in production that will send the logs to S3.  I looked around and it looks like other scripts that use this call do not use any prefixes so I'll keep this as-is for now to be consistent.

Reply:
👍